wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/Plugin.java

```diff
@@ -123,135 +120,56 @@ public OutputStream getOutputStream() {
     }
   }
 
-  abstract static class AbstractContext implements CodeGenerator.Context {
-
-    private class GeneratorResponseFileWriter extends StringWriter {
-
-      private final CodeGeneratorResponse.File.Builder file;
-
-      protected GeneratorResponseFileWriter(
-        CodeGeneratorResponse.File.Builder file) {
-        super();
-        this.file = file;
-      }
+  public static class DescriptorSource {
```

Comment:
This is new and needed by options -- if there is a custom version of options messages being used (e.g. custom `descriptor.proto` included in the operation), then we need to use that.